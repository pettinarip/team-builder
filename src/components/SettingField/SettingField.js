import React from 'react'
import PropTypes from 'prop-types'

const SettingField = ({ title, children }) => (
  <div className={title ? 'settings-field-wrapper label-pad' : 'settings-field-wrapper'}>
    {title && <label className='settings-field-heading'>{title}</label>}
    <div className='settings-field'>
      {children}
    </div>
  </div>
)

SettingField.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default SettingField
