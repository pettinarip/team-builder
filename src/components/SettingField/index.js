import SettingField from './SettingField'

export default SettingField
