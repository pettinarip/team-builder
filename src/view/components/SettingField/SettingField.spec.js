import React from 'react'
import { shallow } from 'enzyme'
import SettingField from './SettingField'

describe('SettingField component', () => {
  it('should render the component properly', () => {
    const component = mountSettingField()
    expect(component).toMatchSnapshot()
  })

  it('should render title text and add a padding class', () => {
    const component = mountSettingField({
      title: 'Testing title!'
    })
    expect(component).toMatchSnapshot()
  })

  it('should render children node', () => {
    const component = mountSettingField({
      children: <div />
    })
    expect(component).toMatchSnapshot()
  })
})

function mountSettingField (props) {
  const propsToUse = {
    title: '',
    children: null,
    ...props
  }

  return shallow(<SettingField {...propsToUse} />)
}
