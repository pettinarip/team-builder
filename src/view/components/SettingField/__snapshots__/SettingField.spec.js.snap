// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingField component should render children node 1`] = `
<div
  className="settings-field-wrapper"
>
  <div
    className="settings-field"
  >
    <div />
  </div>
</div>
`;

exports[`SettingField component should render the component properly 1`] = `
<div
  className="settings-field-wrapper"
>
  <div
    className="settings-field"
  />
</div>
`;

exports[`SettingField component should render title text and add a padding class 1`] = `
<div
  className="settings-field-wrapper label-pad"
>
  <label
    className="settings-field-heading"
  >
    Testing title!
  </label>
  <div
    className="settings-field"
  />
</div>
`;
